foreman/data_refinery_foreman/surveyor/test_end_to_end.py

```diff
@@ -38,10 +39,99 @@
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680048/suppl/GSM2680048_WD1_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680048_WD1_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680049/suppl/GSM2680049_WD2_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680049_WD2_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680050/suppl/GSM2680050_WD3_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680050_WD3_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680051/suppl/GSM2680051_WD4_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680051_WD4_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680052/suppl/GSM2680052_WD6_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680052_WD6_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680053/suppl/GSM2680053_WCu1_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680053_WCu1_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680054/suppl/GSM2680054_WCu2_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680054_WCu2_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680055/suppl/GSM2680055_WCu3_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680055_WCu3_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680056/suppl/GSM2680056_WCu7_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680056_WCu7_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680057/suppl/GSM2680057_WCu8_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680057_WCu8_Cu.CEL.gz",  # noqa
+}
+
+
+def build_surveyor_init_mock(source_type):
+    if source_type == "ARRAY_EXPRESS":
+        original_surveyor = ORIGINAL_ARRAY_EXPRESS_SURVEYOR
+    if source_type == "SRA":
+        original_surveyor = ORIGINAL_SRA_SURVEYOR
+    if source_type == "TRANSCRIPTOME_INDEX":
+        original_surveyor = ORIGINAL_TRANSCRIPTOME_SURVEYOR
+    if source_type == "GEO":
+        original_surveyor = ORIGINAL_GEO_SURVEYOR
+
+    def mock_init_surveyor(survey_job):
+        ret_value = original_surveyor(survey_job)
+
+        def mock_queue_downloader_job_for_original_files(
+            original_files, experiment_accession_code: str = None, is_transcriptome: bool = False,
+        ):
+            for original_file in original_files:
+                original_file.source_url = EXTERNAL_FILE_URL_MAPPING[original_file.source_url]
```

Comment:
`EXTERNAL_FILE_URL_MAPPING[original_file.source_url]` will probably fail if we a new url is found. Can we throw a more meaningful error? And maybe point the developer to some instructions on how to fix it?

Reply:
Good call :+1: